src/Components/Import/Import.tsx

```diff
@@ -1,32 +1,178 @@
-import React, {useState} from 'react';
+import React, { useState } from "react";
+import ReactPlayer from "react-player";
+import {
+  isImage,
+  isVideo,
+  ErrorModal,
+  showError,
+  trimmedName,
+  Media,
+  AddMedia
+} from "..";
+import styles from "./Import.module.css";
+import character from "../../Media/character.png";
+import { Button } from "react-bootstrap";
 
-interface Person {
-  firstName: string;
-}
+type Props = {
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  medias: Media[];
+  setMedias: React.Dispatch<React.SetStateAction<Media[]>>;
+  removeFile: (index: number) => void;
+  addMedia: () => void;
+  addFile: (newMedia: Media[]) => void;
+};
 
-interface Props {
-  text: string;
-  ok?: boolean;
-  i?: number;
-  fn?:(bob: string) => string;
-  person: Person;
-}
+const Import: React.FC<Props> = ({
+  setShow,
+  medias,
+  setMedias,
+  removeFile,
+  addMedia,
+  addFile,
+}) => {
+  const [onDragState, setOnDragState] = useState<boolean>(false);
+  const [onDropState, setOnDropState] = useState<boolean>(false);
 
-interface TextNode {
-  text: string
-}
+  const mediaElement: (addFiles: Media[], file: File) => void = (
+    addFiles,
+    file: File
+  ) => {
+    if (isImage(file)) {
+      let el: JSX.Element = (
+        <img
+          className={styles.renderMedia}
+          src={URL.createObjectURL(file)}
+          onLoad={addMedia}
+        />
+      );
+      let newMedia = new Media(file.name, "image", el);
+      addFiles.push(newMedia);
+    } else if (isVideo(file)) {
+      let el: JSX.Element = (
+        <ReactPlayer
+          url={URL.createObjectURL(file)}
+          width="100%"
+          height="50%"
+          playing={true}
+          onError={() => alert(file + " is unable to play")}
+          id={file.name}
+          onReady={addMedia}
+        />
+      );
+      let newMedia = new Media(file.name, "video", el);
+      addFiles.push(newMedia);
+    } else {
+      showError("invalid file " + file.name);
+    }
+  };
+  const MAXLEN = 30;
 
+  const dropHandler = (e: React.DragEvent<HTMLDivElement>) => {
+    setOnDropState(true);
+    e.preventDefault();
+    const dup: Media[] = [];
+
+    if (e.dataTransfer.items) {
+      // Use DataTransferItemList interface to access the file(s)
+      for (let i = 0; i < e.dataTransfer.items.length; i++) {
+        // If dropped items aren't files, reject them
+        if (e.dataTransfer.items[i].kind === "file") {
+          let file = e.dataTransfer.items[i].getAsFile();
+          if (file == null) {
+            return;
+          }
+          setOnDropState(true);
```

Comment:
does `setOnDropState` ever get set to `false`?

Reply:
setOnDropState implies whether a file is dropped or not - it is initialised to false, and set to "always" true when a file is dropped